hedy.py

```diff
@@ -2832,6 +2816,176 @@ def print_empty_brackets(self, meta, args):
         return self.print(meta, args)
 
 
+@v_args(meta=True)
+@hedy_transpiler(level=1, microbit=True)
+@source_map_transformer(source_map)
+class MicrobitConvertToPython_1(ConvertToPython_1):
+    def print(self, meta, args):
+        # escape needed characters
+        argument = process_characters_needing_escape(args[0])
+        return f"display.scroll('{argument}')"
+
+
+@v_args(meta=True)
+@hedy_transpiler(level=2, microbit=True)
+@source_map_transformer(source_map)
+class MicrobitConvertToPython_2(MicrobitConvertToPython_1, ConvertToPython_2):
+
+    def print(self, meta, args):
+        args_new = [self.make_print_ask_arg(a, meta) for a in args]
+        argument_string = ' '.join(args_new)
+        return f"display.scroll({argument_string})"
+
+    def make_print_ask_arg(self, arg, meta, var_to_escape=''):
+        if self.is_variable(arg, meta.line):
+            return arg
+        else:
+            # If the argument is not a variable, return it as a string literal with quotes
+            return f"'{arg}'"
+
+    def assign(self, meta, args):
+        variable_name = args[0]
+        value = args[1]
+
+        exception = self.make_index_error_check_if_list([value])
+        if self.is_variable(value, meta.line):
+            # if the assigned value is a variable, this is a reassign
+            value = self.process_variable(value, meta.line)
+        else:
+            # if it is not a variable, check if it is a number
+            if value.isdigit() or (value.startswith('-') and value[1:].isdigit()):  # check for integers
+                pass  # do not encapsulate in quotes if it's a number
+            else:
+                try:
+                    float(value)  # check for floats
+                except ValueError:
+                    # encapsulate in quotes if it's not a number
+                    value = f"'{process_characters_needing_escape(value)}'"
+        return exception + variable_name + " = " + value + self.add_debug_breakpoint()
+
+    def sleep(self, meta, args):
+        if not args:
+            return f"sleep(1000){self.add_debug_breakpoint()}"  # Default 1-second sleep in milliseconds
+        else:
+            variable = args[0]
+            if self.is_int(variable):
+                # Direct conversion of seconds to milliseconds
+                milliseconds = int(variable) * 1000
+            else:
+                milliseconds = f"{variable} * 1000"
```

Comment:
If this is a variable (and not an int), you need to register its usage. Otherwise, the tests will fail with an unused variable. Use this code to register a variable usage: `self.add_variable_access_location(variable, meta.line)`

Reply:
Great job, now the three tests are passing all levels. Thank you for all the help.